distributed-testing/src/main/groovy/net/corda/testing/BucketingAllocator.java

```diff
@@ -0,0 +1,203 @@
+package net.corda.testing;
+
+//Why Java?! because sometimes types are useful.
+
+import groovy.lang.Tuple2;
+import org.gradle.api.tasks.TaskAction;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static net.corda.testing.ListTests.DISTRIBUTION_PROPERTY;
+
+public class BucketingAllocator {
+    private static final Logger LOG = LoggerFactory.getLogger(BucketingAllocator.class);
```

Comment:
Gradle's `DefaultTask` already has a `getLogger()` function.

Reply:
Removed the Logger as it was not used